### PR TITLE
[MAINTENANCE] Updated approach to batch config

### PR DESCRIPTION
**Pull request by tyler-hoffman**
**Overview**
Looking for feedback on if we like the flow here. The idea is that the BatchConfig doesn't have access to the context, it doesn't use the project manager's stores, etc; it just calls into the asset's persistence mechanism (to be implemented in the next PR),

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/datasource/fluent/interfaces.py [hunk 1]
```diff
@@ -211,6 +214,11 @@ def batch_request_options(self) -> tuple[str, ...]:
             """One needs to implement "batch_request_options" on a DataAsset subclass."""
         )
 
+    def add_batch_config(self, name: str) -> BatchConfig:
```
Comment (tyler-hoffman) [inline great_expectations/datasource/fluent/interfaces.py:217]: TODO: test this.

File: great_expectations/datasource/fluent/interfaces.py [hunk 2]
```diff
@@ -171,8 +172,10 @@ class DataAsset(FluentBaseModel, Generic[_DatasourceT]):
 
     order_by: List[Sorter] = Field(default_factory=list)
     batch_metadata: BatchMetadata = pydantic.Field(default_factory=dict)
+    batch_configs: List[Any] = Field(default_factory=list)
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/interfaces.py]: Is there a more specific type than `Any` that can be used here?

Comment (tyler-hoffman) [reply great_expectations/datasource/fluent/interfaces.py]: Oh thanks! That was a workaround while I dealt with some circular dependencies. Fixed!

File: great_expectations/datasource/fluent/interfaces.py (hunk 1 above)
Comment (tyler-hoffman) [reply great_expectations/datasource/fluent/interfaces.py:217]: done.

**Review by cdkini — APPROVED** [review]

File: great_expectations/core/batch_config.py [hunk 3]
```diff
@@ -18,24 +19,15 @@ class BatchConfig(pydantic.BaseModel):
...
+    data_asset: DataAsset
 
     def build_batch_request(
         self, batch_request_options: Optional[BatchRequestOptions] = None
     ) -> BatchRequest:
         """Build a BatchRequest from the asset and batch request options."""
         return self.data_asset.build_batch_request(options=batch_request_options)
+
+    def save(self) -> None:
+        self.data_asset._save_batch_config(self)
```
Comment (joshua-stauffer) [inline great_expectations/core/batch_config.py:32]: we're an external caller to the `DataAsset`, i would recommend `DataAsset.save_batch_config` be public (although not `@public_api`).

File: great_expectations/datasource/fluent/interfaces.py [hunk 4]
```diff
@@ -171,8 +172,10 @@ class DataAsset(FluentBaseModel, Generic[_DatasourceT]):
 
     order_by: List[Sorter] = Field(default_factory=list)
     batch_metadata: BatchMetadata = pydantic.Field(default_factory=dict)
+    batch_configs: List[BatchConfig] = Field(default_factory=list)
 
     # non-field private attributes
+    _save_batch_config: Callable[[BatchConfig], None] = pydantic.PrivateAttr()
```
Comment (joshua-stauffer) [inline great_expectations/datasource/fluent/interfaces.py:178]: same comment as above, we want to call this from outside the scope of the asset, right?

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**